consult-yasnippet.el

```diff
@@ -0,0 +1,153 @@
+;;; consult-yasnippet.el --- Provides the command `consult-yasnippet' -*- lexical-binding: t; -*-
+
+;; Author: Mohsin Kaleem, Consult and Selectrum contributors
+;; Maintainer: Daniel Mendler
+;; Created: 2021
+;; License: GPL-3.0-or-later
+;; Version: 0.1
+;; Package-Requires: ((consult "0.1") (yasnippet "0.14") (emacs "26.1"))
+;; Homepage: https://github.com/minad/consult
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Provides the command `consult-yasnippet' which presents all the snippet
+;; expansions available in the current buffer with in-buffer previews.
+;;
+;; This is an extra package, to avoid loading yasnippet in the base consult
+;; package..
+
+
+;;; Code:
+
+(require 'consult)
+(require 'yasnippet)
+
+(defun consult-yasnippet--expand-template (template region region-contents)
+  "Expand the yasnippet template TEMPLATE at point."
+  (deactivate-mark)
+  (goto-char (car region))
+  (when (not (string-equal "" region-contents))
+    (push-mark (point))
+    (push-mark (cdr region) nil t))
+
```

Comment:
Needs a comment/explanation: Why are marks pushed?

Reply:
I think it's related to how `yas-expand-snippet` works. If the region is active it seems to overwrite that region with the snippet expansion. When region-contents isn't empty, there was a region active at the start of `consult-yasnippet` so here we restore that region so `yas-expand-snippet` replaces any previous expansions (or the original `region-contents`) with the current snippet template.

I can try adding a comment to that affect.